Source/Core/VideoCommon/GraphicsModSystem/Config/GraphicsModGroup.cpp

```diff
@@ -0,0 +1,208 @@
+#include "Common/StringUtil.h"
+#include "Core/ConfigManager.h"
+
+#include "VideoCommon/GraphicsModSystem/Config/GraphicsMod.h"
+#include "VideoCommon/GraphicsModSystem/Constants.h"
+#include "VideoCommon/HiresTextures.h"
+
+GraphicsModGroupConfig::GraphicsModGroupConfig(const std::string& game_id) : m_game_id(game_id)
+{
+}
+
+GraphicsModGroupConfig::~GraphicsModGroupConfig() = default;
+
+GraphicsModGroupConfig::GraphicsModGroupConfig(const GraphicsModGroupConfig&) = default;
+
+GraphicsModGroupConfig::GraphicsModGroupConfig(GraphicsModGroupConfig&&) = default;
+
+GraphicsModGroupConfig& GraphicsModGroupConfig::operator=(const GraphicsModGroupConfig&) = default;
+
+GraphicsModGroupConfig& GraphicsModGroupConfig::operator=(GraphicsModGroupConfig&&) = default;
+
+void GraphicsModGroupConfig::Load()
+{
+  const std::string file_path = GetPath();
+
+  std::set<std::string> known_pathes;
+  if (File::Exists(file_path))
+  {
+    std::string json_data;
+    if (!File::ReadFileToString(file_path, json_data))
+    {
+      ERROR_LOG_FMT(VIDEO, "Failed to load graphics mod group json file '{}'", file_path);
+      return;
+    }
+
+    picojson::value root;
+    const auto error = picojson::parse(root, json_data);
+
+    if (!error.empty())
+    {
+      ERROR_LOG_FMT(VIDEO,
+                    "Failed to load graphics mod group json file '{}' due to parse error: {}",
+                    file_path, error);
+      return;
+    }
+    if (!root.is<picojson::object>())
+    {
+      ERROR_LOG_FMT(
+          VIDEO,
+          "Failed to load graphics mod group json file '{}' due to root not being an object!",
+          file_path);
+      return;
+    }
+
+    const auto& mods = root.get("mods");
+    if (mods.is<picojson::array>())
+    {
+      for (const auto& mod_json : mods.get<picojson::array>())
+      {
+        if (mod_json.is<picojson::object>())
+        {
+          const auto& mod_json_obj = mod_json.get<picojson::object>();
+          auto graphics_mod = GraphicsModConfig::Create(&mod_json_obj);
+          if (!graphics_mod)
+          {
+            continue;
+          }
+          graphics_mod->DeserializeFromProfile(mod_json_obj);
+
+          auto mod_full_path = graphics_mod->GetAbsolutePath();
+#ifdef _WIN32
+          // On Windows, replace all '\' with '/' since we assume the latter in various places in
+          // the codebase.
+          for (char& c : mod_full_path)
+          {
+            if (c == '\\')
+              c = '/';
+          }
+#endif
+          known_pathes.insert(std::move(mod_full_path));
+          m_graphics_mods.push_back(*graphics_mod);
+        }
+      }
+    }
+  }
+
+  const std::set<std::string> graphics_mod_user_directories =
+      GetTextureDirectoriesWithGameId(File::GetUserPath(D_GRAPHICSMOD_IDX), m_game_id);
+
+  for (const auto& graphics_mod_directory : graphics_mod_user_directories)
+  {
+    const auto json_files = Common::DoFileSearch({graphics_mod_directory}, {"metadata.json"});
```

Comment:
Yeah, this does support `metadata.json` anywhere under root but that's probably not necessary.  I'll go with your solution, thanks!

Reply:
I think it would support `metadata.json` anywhere under root if you specified `true` for `DoFileSearch`'s `recusrive` parameter, but that defaults to `false`.  (I don't think it's particularly necessary either, but it might be useful if the user wants to put multiple graphics mods in the same folder?)